Change logging statements to remove some noise

saifsultanc opened the pull request:
## Description
This enhances some of the logging statements. Following updates are on this branch:
- `get_inbox_form_ids()` on _/includes/class-api.php_ presented in a compact way.
- `get_assignees()` on _/includes/steps/class-step.php_ presented in a compact way.
-  `maybe_process_expiration_and_reminders()` on _/class-gravity-flow.php_ presented in a compact way.
- `send_webhook()` on _/includes/steps/class-step-webhook.php_ enhanced with lesser noise option on request-response.
- `send_webhook()` on _/includes/steps/class-step-webhook.php_ enhanced with lesser noise option on Updated Entry.
- `maybe_display_assignee_status_list()` _/includes/steps/class-step-user-input.php_ enhanced with lesser noise option on assignee details.

To test the lighter logging snippets, define the constant `GF_LIGHT_LOGGING` on _wp-config.php_

`define( 'GF_LIGHT_LOGGING', true );`

## Testing instructions
Review log files for the steps as described.

## Automated Test Enhancements
N/A
 
## Screenshots
N/A

## Documentation Changes?
N/A

## Checklist:
- [x] I've tested the code.
- [x] My code follows the WordPress code style. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code follows the inline documentation standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/ -->

Idealien commented on includes/steps/class-step-webhook.php:
```diff
@@ -688,8 +688,12 @@ function send_webhook() {
 		$response = wp_remote_request( $url, $args );
-		$this->log_debug( __METHOD__ . '() - request: ' . print_r( $args, true ) );
-		$this->log_debug( __METHOD__ . '() - response: ' . print_r( $response, true ) );
+		$args_adjusted = $args;
```
Naming as _adjusted doesn't convey what the purpose to the variables are.
Would $log_args / $log_response be better?

saifsultanc replied:
@Idealien Yes, updated naming.

Idealien commented on includes/steps/class-step-webhook.php:
```diff
@@ -688,8 +688,12 @@ function send_webhook() {
+		$args_adjusted = $args;
+		unset( $args_adjusted['body'] );
+		$response_adjusted = $response;
+		unset( $response_adjusted['headers'], $response_adjusted['cookies'], $response_adjusted['http_response'] );
```
I agree that this makes the logging much cleaner.

There may be cases where receiving the unset values for debugging purposes is useful.  Do you think there will be many scenarios that "light logging" is to be the default? Perhaps a constant to check whether logging level/scenario warrants the more verbose (existing minus encoding format changes) version?

saifsultanc replied:
@Idealien Yes, I've added the GF_LIGHT_LOGGING. If that's set to true via wp-config.php, then we process that "light logging" scenarios.

saifsultanc commented:
Summary.

1. Added the LIGHT LOGGING constant option.
2. Updated send_webhook() request-response to have the light logging and normal versions.
3. Added the lesser noise option on Update Entry, also on send_webhook().
4. Added the lesser noise option on assignee list, printed on maybe_display_assignee_status_list().

Let me know if that looks better, or any other way we can improve these.
Also, any other suggestions on logging updates on other files?

Thanks!

Idealien commented on includes/steps/class-step-webhook.php:
```diff
@@ -688,8 +688,12 @@ function send_webhook() {
+		$args_adjusted = $args;
+		unset( $args_adjusted['body'] );
+		$response_adjusted = $response;
+		unset( $response_adjusted['headers'], $response_adjusted['cookies'], $response_adjusted['http_response'] );
```
Additional checks are needed here. If a WP_Error is returned, this causes an error.
Same on the $args_adjusted['body'] although much less likely.

saifsultanc replied:
Fixed.
